src/store.js

```diff
@@ -0,0 +1,94 @@
+import { reactive } from 'vue'
+
+import {
+  POLL_NUMBER_DETECTION_PATTERN,
+  POLL_QUOTED_PARAMETER_DETECTION_PATTERN,
+  POLL_QUOTED_PARAMETER_EXTRACTION_PATTERN,
+  POLL_SIMPLE_DETECTION_PATTERN,
+  POLL_VOTE_EXTRACTION_PATTERN
+} from './messageCheckers.js'
+
+const DEBUG_POLL_STATE = {
+  active: true,
+  visible: true,
+  title: 'Debug Mode Poll',
+  options: { 1: 'Pizza', 2: 'Jam', 3: 'Coffee' },
+  userVotes: { user1: '1', user2: '3', user3: '3' },
+  position: 'tl'
+}
+
+const INITIAL_STATE = {
+  active: false,
+  visible: false,
+  title: 'Poll',
+  options: {},
+  userVotes: {},
+  position: 'tl'
+}
+
+export default reactive({
+  ...INITIAL_STATE,
+  stopPoll() {
+    this.active = false
+  },
+  resumePoll() {
+    this.active = true
+  },
+  endPoll() {
+    Object.assign(this, INITIAL_STATE)
+  },
+  resetPoll() {
+    this.active = true
+    this.userVotes = {}
+  },
+  updatePollTitle(message) {
+    const options = message.match(POLL_QUOTED_PARAMETER_EXTRACTION_PATTERN)
+    const title = options.shift().replaceAll('"', '')
+    this.title = title
+  },
+  updatePosition(message) {
+    const positionSuffix = message.split('_')[1]
+    this.position = positionSuffix
+  },
+  castVote(message, username) {
+    // don't cast vote if poll is invisible or inactive
+    if (!this.visible || !this.active) return
+
+    const voteNumber = message.match(POLL_VOTE_EXTRACTION_PATTERN)?.[1]
+
+    if (voteNumber && (this.options[voteNumber] || voteNumber === '0')) {
+      this.userVotes[username] = voteNumber
+    }
+  },
+  setDebugMode(debugMode) {
+    this.debugMode = debugMode
+    Object.assign(this, DEBUG_POLL_STATE)
+  },
+  startPoll(message) {
+    // don't do anything if the a poll is still running
+    if (this.visible || this.active) return
+
```

Comment:
Das alte `handlePollStart` hatte hier eine vollständige State-Zuweisung, die alles zurückgesetzt hat. Analog könnte man hier jetzt
```js
Object.assign(this, {
  active: true,
  visible: true,
  title: "Poll",
  options: {},
  userVotes: {},
});
```
machen.

Reply:
Ja, genau das fehlte auch noch. Ich hab das jetzt einfach direkt auf this gesetzt an der Stelle.

```js
this.options = {}
this.userVotes = {}
this.active = true
this.visible = true
this.title = 'Poll'
```